include/common/serial_port.h

```diff
@@ -0,0 +1,13 @@
+#ifndef COMMON_SERIAL_PORT
+#define COMMON_SERIAL_PORT
+
+typedef struct {
+  void (*init)(void);
+  void (*putchar)(int);
+} SerialPrint;
+
+extern SerialPrint platform_serial;
```

Comment:
I think it would be simpler to call this "serial_port".
```
extern SerialPort serial_port;
```

I know what you mean by platform but I don't think it's needed, we can see from context it's an abstraction.